dspace-api/src/main/java/org/dspace/app/ldn/service/impl/LDNMessageServiceImpl.java

```diff
@@ -0,0 +1,364 @@
+                        log.info(
+                            "No processor found for LDN message " + msgs.get(i));
+                        msg.setQueueStatus(LDNMessageEntity.QUEUE_STATUS_UNMAPPED_ACTION);
+                        msg.setQueueAttempts(msg.getQueueAttempts() + 1);
+                        update(context, msg);
+                    }
+                }
+                if (processor != null) {
+                    try {
+                        msg.setQueueLastStartTime(new Date());
+                        msg.setQueueStatus(LDNMessageEntity.QUEUE_STATUS_PROCESSING);
+                        msg.setQueueTimeout(DateUtils.addMinutes(new Date(), timeoutInMinutes));
+                        update(context, msg);
+                        ObjectMapper mapper = new ObjectMapper();
+                        Notification notification = mapper.readValue(msg.getMessage(), Notification.class);
+                        processor.process(context, notification);
+                        msg.setQueueStatus(LDNMessageEntity.QUEUE_STATUS_PROCESSED);
+                        result = 1;
+                    } catch (Exception e) {
+                        result = -1;
+                        log.error(e);
+                    } finally {
+                        msg.setQueueAttempts(msg.getQueueAttempts() + 1);
+                        update(context, msg);
+                    }
+                } else {
+                    log.info("Found x" + msgs.size() + " LDN messages but none processor found.");
+                }
+            }
+        } catch (SQLException e) {
+            result = -1;
+            log.error(e);
+        }
+        return result;
+    }
+
+    @Override
+    public int checkQueueMessageTimeout(Context context) {
+        int result = 0;
+        int timeoutInMinutes = configurationService.getIntProperty("ldn.processor.queue.msg.timeout");
+        if (timeoutInMinutes == 0) {
+            timeoutInMinutes = 60;
+        }
+        int maxAttempts = configurationService.getIntProperty("ldn.processor.max.attempts");
+        if (maxAttempts == 0) {
+            maxAttempts = 5;
+        }
+        log.debug("Using parameters: [timeoutInMinutes]=" + timeoutInMinutes + ",[maxAttempts]=" + maxAttempts);
+        /*
+         * CST-10631 put failed on processing messages with timed-out timeout and
+         * attempts >= configured_max_attempts put queue on processing messages with
+         * timed-out timeout and attempts < configured_max_attempts
+         */
+        List<LDNMessageEntity> msgsToCheck = null;
+        try {
+            msgsToCheck = findProcessingTimedoutMessages(context);
+        } catch (SQLException e) {
+            result = -1;
+            log.error("An error occured on searching for timedout LDN messages!", e);
+            return result;
+        }
+        if (msgsToCheck == null || msgsToCheck.isEmpty()) {
+            log.info("No timedout LDN messages found in queue.");
```

Comment:
I don't think we should log this as an INFO message. It doesn't seem useful to know that "no timed out messages were found".  It's only useful if we *find* timed out messages.  Currently, I'm finding that every hour I see this in the logs:
```
2024-02-16 22:00:00,024 INFO  unknown unknown org.dspace.app.ldn.service.impl.LDNMessageServiceImpl @ No timedout LDN messages found in queue.
```